src/app/shell/result/result.component.ts

```diff
@@ -58,17 +62,29 @@ export class ResultComponent implements OnInit, OnDestroy {
   constructor(
     private actions$: Actions,
     private store: Store,
-    public navigationBarService: NavigationBarService
+    public navigationBarService: NavigationBarService,
+    private route: ActivatedRoute,
+    private router: Router,
+
   ) { }
 
   ngOnInit(): void {
 
+    this.route.params.subscribe(params => {
+      if (params.param === 'map') {
+        this.currentView = this.viewType.map
+      } else {
+        this.currentView = this.viewType.data
+      }
```

Comment:
this.currentView = params.param === 'map' ? this.viewType.map : this.viewType.data;